test/test_inventory.py

```diff
@@ -0,0 +1,129 @@
+from unittest.mock import MagicMock
+from asynctest import CoroutineMock
+import pytest
+import csv
+import HAL9666.AuctionMasterBot as bot
+import io
+
+@pytest.mark.asyncio
+async def test_inventory():
+
+    general_csv_file = create_csv(
+        [
+            {"Username":"Kindling", "Ticker": "C", "Amount":"200", "NaturalId": "UV-351a"}
+        ]
+    )
+
+    shipyard_csv_file = create_csv(
+        [
+            {"Username":"Felmer", "Ticker": "WCB", "Amount":"3", "NaturalId": "UV-351a"}
+        ]
+    ) 
+    
+    fake_fio_response = MagicMock()
+    fake_fio_response.status_code = 200
+    fake_fio_response.text = general_csv_file
+
+    bot.requests.get = MagicMock(return_value=fake_fio_response)
+
+    bot.getSellerData = MagicMock(return_value = {"Kindling":[], "Felmer": [], "Gilith": []})
+
+    inv = await bot.whohas(CoroutineMock(), "C")
+    assert inv[0] == ("Kindling", 200)
+
+    general_csv_file = create_csv(
+        [
+            {"Username":"Kindling", "Ticker": "C", "Amount":"200", "NaturalId": "UV-351a"},
+            {"Username":"Felmer", "Ticker": "C", "Amount":"250", "NaturalId": "UV-351a"}
+        ]
+    )
+    fake_fio_response.text = general_csv_file
+
+    inv = await bot.whohas(CoroutineMock(), "C")
+
+    assert inv[0] == ("Felmer", 250)
+    assert inv[1] == ("Kindling", 200)
+
+    fake_fio_response.status_code = 500
+
+    ctx = MagicMock()
+    ctx.reply = CoroutineMock()
+    inv = await bot.whohas(ctx, "C")
+
+    assert inv[0] == ("Felmer", 250)
+    assert inv[1] == ("Kindling", 200)
+
+    fake_fio_response.status_code = 200
+    fake_fio_response.text = shipyard_csv_file
+    inv = await bot.whohas(ctx, "WCB")
+
+    assert len(inv) == 1
+    assert inv[0] == ("Felmer", 3)
+
+    fake_fio_response.status_code = 500
+    inv = await bot.whohas(ctx, "C")
+
+    assert inv[0] == ("Felmer", 250)
+    assert inv[1] == ("Kindling", 200)
+
+@pytest.mark.asyncio
+async def test_pos_filter():
+
+    # when someone has set POS filter, we should only count the amounts from those locations
+    bot.getSellerData = MagicMock(return_value = {"Kindling":["UV-351a"], "Felmer": ["XG-521b"], "Gilith": []})
+
+    csv_file = create_csv(
+        [
+            {"Username":"Kindling", "Ticker": "C", "Amount":"200", "NaturalId": "UV-351a"},
+            {"Username":"Kindling", "Ticker": "C", "Amount":"100", "NaturalId": "KW-688c"},
+            {"Username":"Gilith", "Ticker": "C", "Amount":"100", "NaturalId": "UV-351a"},
+            {"Username":"Felmer", "Ticker": "C", "Amount":"250", "NaturalId": "UV-351a"}
+        ]
+    )
+    
+    fio_response = MagicMock()
+    fio_response.status_code = 200
+    fio_response.text = csv_file
+
+    bot.requests.get = MagicMock(return_value=fio_response)
+
+    inv = await bot.whohas(MagicMock(), "C", False)
+
+    assert len(inv) == 2
+    assert inv[0] == ("Kindling", 200)
+    assert inv[1] == ("Gilith", 100)
```

Comment:
maybe change this to:
assert ("Kindling", 200) in inv
we shouldn't rely on order, it might change in the future if we sort the results differently
this also applies to test_inventory